### PR TITLE
Port current_currency helper module from spree_multi_domain

### PR DESCRIPTION
**Pull request by mamhoff**
Previously, the `current_currency` helper would always return the
configured default currency for the Solidus installation. Solidus core
allows setting a default currency per `Spree::Store`, and this commit
adds a `current_currency` helper that respects that setting.

This does purposefully not include setting a currency via the session
(like the `spree_multi_currency` gem does. Solidus core does not allow
this, and it is much less complex to switch currencies with the configured
store.

This PR will also remove duplicate helper definitions from frontend and core.

### CODE REVIEW

File: core/lib/spree/core/controller_helpers/pricing.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    module ControllerHelpers
+      module Pricing
+        extend ActiveSupport::Concern
+
+        included do
+          helper_method :current_currency
+        end
+
+        def current_currency
+          current_store.try(:default_currency).presence || Spree::Config[:currency]
```
Comment (gmacdougall) [inline core/lib/spree/core/controller_helpers/pricing.rb]: Do we expect current the current_store default currency to return a non-nil blank value (like `""`)? If not, we can simplify this to:

```Ruby
current_store.try!(:default_currency) || Spree::Config[:currency]
```

**Comment by gmacdougall** [comment]
This is looking good so far.

File: core/lib/spree/core/controller_helpers/pricing.rb (hunk 1 above)
Comment (mamhoff) [inline core/lib/spree/core/controller_helpers/pricing.rb]: I took some specs from `solidus_multi_domain`, and this case has been explicitly tested for: https://github.com/solidusio/solidus/pull/1060/files#diff-14ee7de4c10a42a47306354bec6b902eR20

**Comment by mamhoff** [comment]
I added the `try!` you suggested; otherwise I think this can go in as-is (unless you insist on removing the `presence` call).

**Comment by gmacdougall** [comment]
:+1:

**Comment by jhawthorn** [comment]
:+1:

I think we'll want a CHANGELOG entry as this is a (beneficial) behaviour change that we now respect the `default_currency` option on a store

**Comment by mamhoff** [comment]
@jhawthorn Added Changelog :)

**Pull request merged**